lib/class-wp-rest-block-types-controller.php

```diff
@@ -0,0 +1,376 @@
+				$block_namespace = $pieces[0];
+				if ( $namespace !== $block_namespace ) {
+					continue;
+				}
+			}
+			$data[ $obj->name ] = $this->prepare_response_for_collection( $block_type );
+		}
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Checks if a given request has access to read a block type.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|bool True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		$check = $this->check_read_permission();
+		if ( ! $check ) {
+			return new WP_Error( 'rest_cannot_read_block_type', __( 'Cannot view block type.', 'gutenberg' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Checks whether a given block type should be visible.
+	 *
+	 * @return WP_Error|bool True if the block type is visible, otherwise false.
+	 */
+	protected function check_read_permission() {
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return new WP_Error( 'rest_cannot_view', __( 'Sorry, you are not allowed to manage block types.', 'gutenberg' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get the block, if the name is valid.
+	 *
+	 * @param string $name Block name.
+	 * @return WP_Block_Type|WP_Error Block type object if name is valid, WP_Error otherwise.
+	 */
+	protected function get_block( $name ) {
+		$block_type = $this->registry->get_registered( $name );
+		if ( empty( $block_type ) ) {
+			return new WP_Error( 'rest_block_type_invalid', __( 'Invalid block type.', 'gutenberg' ), array( 'status' => 404 ) );
+		}
+
+		return $block_type;
+	}
+
+	/**
+	 * Retrieves a specific block type.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return WP_Error|WP_REST_Response Response object on success, or WP_Error object on failure.
+	 */
+	public function get_item( $request ) {
+		$block_name = sprintf( '%s/%s', $request['namespace'], $request['name'] );
+		$block_type = $this->get_block( $block_name );
+		if ( is_wp_error( $block_type ) ) {
+			return $block_type;
+		}
+		$data = $this->prepare_item_for_response( $block_type, $request );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepares a block type object for serialization.
+	 *
+	 * @param WP_Block_Type   $block_type block type data.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 *
+	 * @return WP_REST_Response block type data.
+	 */
+	public function prepare_item_for_response( $block_type, $request ) {
+
+		$fields = $this->get_fields_for_response( $request );
+		$data   = array();
+
+		if ( in_array( 'attributes', $fields, true ) ) {
```

Comment:
```suggestion
		if ( rest_is_field_included( 'attributes', $fields ) ) {
```

Reply:
This and similar changed would require 5.3+. This plugin can be installed on 5.2 or below. Let's make this change when we work on a core patch.

Reply:
Gutenberg currently requires 5.3+

https://github.com/WordPress/gutenberg/blob/master/gutenberg.php#L147